consensus/polybft/contractsapi/init.go

```diff
@@ -20,10 +18,9 @@ var (
-	L1ExitDeployedBytecode = hex.MustDecodeHex("0x608060405234801561001057600080fd5b50600436106100575760003560e01c806361bc221a1461005c57806373d4a13a14610078578063767800de1461008d578063af640d0f146100b8578063f43cda8b146100c1575b600080fd5b61006560035481565b6040519081526020015b60405180910390f35b6100806100d6565b60405161006f91906101ab565b6001546100a0906001600160a01b031681565b6040516001600160a01b03909116815260200161006f565b61006560005481565b6100d46100cf36600461020f565b610164565b005b600280546100e3906102e8565b80601f016020809104026020016040519081016040528092919081815260200182805461010f906102e8565b801561015c5780601f106101315761010080835404028352916020019161015c565b820191906000526020600020905b81548152906001019060200180831161013f57829003601f168201915b505050505081565b6000839055600180546001600160a01b0319166001600160a01b03841617905560026101908282610371565b50600380549060006101a183610431565b9190505550505050565b600060208083528351808285015260005b818110156101d8578581018301518582016040015282016101bc565b506000604082860101526040601f19601f8301168501019250505092915050565b634e487b7160e01b600052604160045260246000fd5b60008060006060848603121561022457600080fd5b8335925060208401356001600160a01b038116811461024257600080fd5b9150604084013567ffffffffffffffff8082111561025f57600080fd5b818601915086601f83011261027357600080fd5b813581811115610285576102856101f9565b604051601f8201601f19908116603f011681019083821181831017156102ad576102ad6101f9565b816040528281528960208487010111156102c657600080fd5b8260208601602083013760006020848301015280955050505050509250925092565b600181811c908216806102fc57607f821691505b60208210810361031c57634e487b7160e01b600052602260045260246000fd5b50919050565b601f82111561036c57600081815260208120601f850160051c810160208610156103495750805b601f850160051c820191505b8181101561036857828155600101610355565b5050505b505050565b815167ffffffffffffffff81111561038b5761038b6101f9565b61039f8161039984546102e8565b84610322565b602080601f8311600181146103d457600084156103bc5750858301515b600019600386901b1c1916600185901b178555610368565b600085815260208120601f198616915b82811015610403578886015182559484019460019091019084016103e4565b50858210156104215787850151600019600388901b60f8161c191681555b5050505050600190811b01905550565b60006001820161045157634e487b7160e01b600052601160045260246000fd5b506001019056fea264697066735822122038ac4dcd21701f192702ee4a99e63fef6c90075b018b5338d65c1d477a999c6364736f6c63430008110033")                                                                 //nolint
-
-	L1ExitTestABI *abi.ABI
+	//go:embed test-contracts/*
+	testContracts       embed.FS
+	TestL1StateReceiver *artifact.Artifact
```

Comment:
We are not using `embed` with the other artifacts right? I would prefer to standardize that process and not introduce a different workflow. Note @goran-ethernal is also working on parallel on some of this work.

Reply:
Note that I would prefer to use `embed` too.

Reply:
> We are not using embed with the other artifacts right?

Note that there is one big difference between `TestL1StateReceiver` contract and all the others: `TestL1StateReceiver` is not part of `core-contracts` repo and it is only there to implement missing part in logic of smart contracts (https://github.com/0xPolygon/core-contracts/blob/dev/contracts/root/ExitHelper.sol#L88, as we currently don't have "production" SC counter-part which implements `onL2StateReceived` function). Therefore Boris decided to implement this part as this dummy contract which is going to be used only for testing purposes. Also that contract is not expected to be changed and it is not part of auto generation process and `gen_sc_data.go` (as other smart contracts).

So what we could do (although workflow would still not be the same) is to copy all the JSON artifacts we have in the `core-contracts` into this `test-contracts` folder (although probably rename it to simply `contracts`) and load appropriate JSONs the same way we are doing with `TestL1StateReceiver`. Does that make sense?

Reply:
> Note @goran-ethernal is also working on parallel on some of this work.

Yep I'm aware of it. I think these changes aren't in collision with his work 🙂